File: 1099-two-sum-less-than-k/1099-two-sum-less-than-k.cpp
class Solution {
public:
    int twoSumLessThanK(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int x = -1, i = 0, j = nums.size() - 1;
        while(i < j) {
            if(nums[i] + nums[j] < k)
                x = max(x, nums[i++] + nums[j++]);
            j--;
        }
        return x;
    }
};

File: 1099-two-sum-less-than-k/1099-two-sum-less-than-k.py
class Solution:
    def twoSumLessThanK(self, nums: List[int], k: int) -> int:
        nums.sort()
        x = -1
        i, j = 0, len(nums) - 1
        while i < j:
            if nums[i] + nums[j] < k:
                x = max(x, nums[i] + nums[j])
                i += 1
            else:
                j -= 1
            
        return x